src/main/kotlin/no/nav/amt/aktivitetskort/repositories/DeltakerRepository.kt

```diff
@@ -48,7 +52,14 @@ class DeltakerRepository(
 			return RepositoryResult.NoChange()
 		}
 
-		if (deltaker == old?.deltaker && !skalRelasteDeltakere) return RepositoryResult.NoChange()
+		if (deltaker == old?.deltaker) return RepositoryResult.NoChange()
+
+		if (deltaker.status.type in AVSLUTTENDE_STATUSER && deltaker.sluttdato?.isBefore(lanseringAktivitetsplan) == true &&
+			(old?.deltaker == null || old.deltaker.status.type in AVSLUTTENDE_STATUSER)
```

Comment:
Det er ikke helt tydelig (les: jeg forstår ikke 😅) hvorfor vi må sjekke på old.deltaker sin status. Er det fordi vi allerede har tatt over et aktivitetskort vi egentlig ikke skulle?

Reply:
Hvis en deltaker ble opprettet før aktivitetsplanen ble lansert og har en gjennomførende status så skal det opprettes aktivitetskort hvis det skjer endringer på deltakeren. Derfor kan det hende at vi oppretter kort for deltakere som startet før aktivitetsplanen ble lansert. Hvis man av en eller annen grunn oppdager i ettertid at denne deltakelsen egentlig ble avsluttet før aktivitetsplanen ble opprettet så ønsker vi at et evt kort skal settes til avsluttende status. Se svar fra Lars her: https://nav-it.slack.com/archives/C02CFP3TRPE/p1705303731301419?thread_ts=1705067103.364069&cid=C02CFP3TRPE

Reply:
Hvis statusen vi hadde fra før uansett var avsluttende trenger vi ikke endre noe på et evt kort, derfor sjekker jeg statusen :)

Reply:
Jeg regnet med det var en grunn, det er bare litt vanskelig å forstå hvorfor det er slik i koden, kan det beskrives med en variabler, eller hvis ikke - en kommentar?

Reply:
Jeg kan prøve å omformulere koden litt - kanskje trekke det ut i en egen funksjon jeg kan navngi :)

Reply:
ble det bedre nå..?

Reply:
Fikk grønt lys for dette på slack :)